storage/engine.go

```diff
@@ -531,6 +531,11 @@ func (e *Engine) DeleteBucket(orgID, bucketID platform.ID) error {
 
 // DeleteBucketRange deletes an entire bucket from the storage engine.
 func (e *Engine) DeleteBucketRange(orgID, bucketID platform.ID, min, max int64) error {
+	// Snapshot to clear the cache to reduce write contention.
+	if err := e.engine.WriteSnapshot(context.Background()); err != nil {
```

Comment:
This is going to abort the deletion if we can't snapshot the cache. Sometimes we can't snapshot the cache for good reasons, but that doesn't mean we should abort a retention check / delete bucket range.

One concrete example of this is if the cache is already snapshotting, in which case `tsm1.ErrSnapshotInProgress` is going to be returned.